snowflake/src/test/java/org/apache/iceberg/snowflake/SnowflakeCatalogTest.java

```diff
@@ -0,0 +1,189 @@
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.snowflake;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URISyntaxException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Scanner;
+import java.util.UUID;
+import org.apache.iceberg.CatalogProperties;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.catalog.Namespace;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.jdbc.JdbcClientPool;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+public class SnowflakeCatalogTest {
+
+  public SnowflakeCatalogTest() {}
+
+  static final String TEST_CATALOG_NAME = "slushLog";
+  static final String TEST_SETUP_SCRIPT = "setup_mock_snowflakedb.sql";
+
+  static final String TEST_CUSTOM_FILE_IO = "org.apache.iceberg.snowflake.TestFileIO";
+  static String uri;
+  static SnowflakeCatalog catalog;
+
+  @ClassRule public static TemporaryFolder folder = new TemporaryFolder();
+
+  @BeforeClass
+  public static void beforeClass() {
+
+    File dbFile;
+    try {
+      dbFile = folder.newFile(UUID.randomUUID().toString().replace("-", "") + ".db");
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+
+    Map<String, String> properties = Maps.newHashMap();
+
+    uri = "jdbc:sqlite:" + dbFile.getAbsolutePath();
+
+    properties.put(CatalogProperties.URI, uri);
+
+    properties.put(CatalogProperties.FILE_IO_IMPL, TEST_CUSTOM_FILE_IO);
+
+    try {
+      Connection con = DriverManager.getConnection(uri);
+      Path path =
+          Paths.get(
+              Objects.requireNonNull(
+                      SnowflakeCatalogTest.class.getClassLoader().getResource(TEST_SETUP_SCRIPT))
+                  .toURI());
+      importSQLScript(con, new FileInputStream(path.toFile()));
+
+      catalog = new SnowflakeCatalog();
+      JdbcClientPool connectionPool = new JdbcClientPool(uri, properties);
+      catalog.setQueryFactory(new TestQueryFactory(connectionPool));
```

Comment:
Since the underlying JDBC here won't be exercising a real Snowflake data model anyways, IMO it's better to keep this unittest contained to the responsibilities of SnowflakeCatalog itself, which is namely simply to translate between our own in-memory data model (entities/) into Iceberg idioms. Then we don't need a fake underlying database or a fake query factory.

We could also try to unittest the query factory (which we should rename to JdbcSnowflakeClient) by fake or mock JdbcClientPools and just verify the queries generated rather than using a fake sqlite database.

Reply:
Done